tempesta_fw/t/unit/dl-182/main.c

```diff
@@ -0,0 +1,82 @@
+#include "http.h"
+#include <sys/time.h>
+#include <assert.h>
+#include <stdio.h>
+
+static inline unsigned long
+tv_to_ms(const struct timeval *tv)
+{
+    return ((unsigned long)tv->tv_sec * 1000000 + tv->tv_usec) / 1000;
+}
+
+int main(int argc, char ** argv)
+{
+    unsigned long iterations = 10;
+    int nfile = 1;
+    TfwPool * pool = (TfwPool*)malloc(sizeof(*pool));
+    assert(pool != NULL);
+
+    TfwHttpReq req;
+
+    if (argc == 1) {
+        printf("Usage: %s [-n iterations] file1 [fileN...]\n");
+        return 0;
+    }
+
+    if (!strcmp(argv[1], "-n")) {
+        if (argc < 4) {
+            printf("Usage: %s [-n iterations] file1 [fileN...]\n");
+            return 1;
+        }
+        iterations = strtoul(argv[2], NULL, 10);
+        nfile = 3;
+    }
+
+    FILE * source = fopen(argv[nfile], "rt");
+    if (!source) {
+        printf("error: input file '%s' cannot be opened\n", argv[nfile]);
+        return 1;
+    }
+
+    fseek(source, 0, SEEK_END);
+    size_t len = ftell(source);
```

Comment:
usually `fstat(2)` is used to get file size. Moreover, since you set file pointer using `fseek()` following `fread()` doesn't read any data and your test doesn't do anything.